lib/mapper.js

```diff
@@ -314,6 +321,34 @@ function formatMetadata(props){
 }
 
 /**
+ * Flatten selectively based on your settings. You can either stringify, flatten, or drop the properties.
+ * Intercom rejects nested objects so you must choose a method.
+ *
+ * @param {Object} obj
+ * @param {Object} blacklisted
+ * @return {Object} ret
+ * @api private
+ */
+
+function formatNested(obj, blacklisted){
+  return foldl(function(attrs, value, key){
+    var trait = blacklisted[key];
+    var pair = {};
+    pair[key] = value;
+
+    if (trait === 'stringify') {
+      attrs[key] = JSON.stringify(value);
+    } else if (trait === 'drop') {
+      return attrs;
+    } else {
+      attrs = extend(flatten(pair), attrs);
```

Comment:
I left it the default because that is the _current_ behavior, everything is flattened.

But good call. I think that would add more flexibility. I'll make the default flatten for everyone via mongo script and expose that UI dropdown option. @tsholmes